source/FreeRTOS_IP.c

```diff
@@ -1544,6 +1544,9 @@ static void prvProcessEthernetPacket( NetworkBufferDescriptor_t * const pxNetwor
     eFrameProcessingResult_t eReturned = eReleaseBuffer;
 
     configASSERT( pxNetworkBuffer != NULL );
+    configASSERT( pxNetworkBuffer->pxInterface != NULL );
+    configASSERT( pxNetworkBuffer->pxEndPoint != NULL );
```

Comment:
If packet processing relies on these non-NULL checks from here, wouldn't it be better to have them as an if check rather than `configASSERT`, since most release configurations don't define `configASSERT`.

Reply:
`configASSERT( pxNetworkBuffer->pxInterface != NULL );` and `configASSERT( pxNetworkBuffer->pxEndPoint != NULL );` should go just below `ipCONSIDER_FRAME_FOR_PROCESSING`. 
`configASSERT( pxNetworkBuffer != NULL );` would be fine if you add an if( pxNetworkBuffer != NULL )` in `prvHandleEthernetPacket`.

Reply:
> `configASSERT( pxNetworkBuffer->pxInterface != NULL );` and `configASSERT( pxNetworkBuffer->pxEndPoint != NULL );` should go just below `ipCONSIDER_FRAME_FOR_PROCESSING`.
If there isn't a valid interface/endpoint associated with an incoming packet, I can't find any reason to do an `ipCONSIDER_FRAME_FOR_PROCESSING` check since most of the checks in `eConsiderFrameForProcessing` are memory comparisons of the MAC address.

Reply:
`eConsiderFrameForProcessing` does the `FreeRTOS_FindEndPointOnMAC` check which I suppose could be a last effort recovery at determining the endpoint if the NetworkInterface didn't set it correctly. Or I guess you could just require it be set anyways.

Reply:
Very often, `configASSERT()` is called just to wake up the developer, so she/he notices that there is a fatal problem.
I always deliver my software with a working version of `configASSERT()`. Up until know I never had an assert.
One alternative is to return `eReleaseBuffer` to the caller and `FreeRTOS_debug_printf` the error.

Reply:
@tony-josi-aws I personally prefer what you are suggesting: do all the NULL checks and cleanly drop the buffer. My reasoning is that if I didn't catch something in testing, it shouldn't break the product in production. To be honest, the only reason I'm adding those asserts is because I'm mimicking what the code already does and I'm trying to follow the original intent. In my personal code, I never do asserts. I always check early, and handle the situation as cleanly as possible and certainly not by going in a forever loop.

Let me try this: Let me substitute the asserts with clean dropping of packets and reexamine where those checks need to go and you can tell me what you think.

Reply:
Since we are removing the `pxEndPoint != NULL` checks in later part of the code, I believe its safer to do the NULL check and drop the packets here.